### PR TITLE
Plane stress migration to tensor mechanics

### LINKED ISSUE #7902: Plane strain migration to tensor mechanics
### Description of the enhancement or error report
migrate the ability to simulate plane stress from solid mechanics to tensor mechanics 

### Rationale for the enhancement or information for reproducing the error
https://groups.google.com/forum/#!searchin/moose-users/plane$20stress%7Csort:relevance/moose-users/VDsurHVXFcU/WRxJwFmVAAAJ

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

### PR DESCRIPTION
**Pull request by hchen139**
closes #7902

### CODE REVIEW

**Comment by hchen139** [comment]
@bwspenc @sapitts This is for the plane stress case in tensor mechanics. The golded result for small deformation test is from solid mechanics but before applying volumetric correction by @sveerara. Otherwise, the results will be quite different from each other.

**Comment by sapitts** [comment]
Thanks for working on this so quickly @hchen139 !

File: modules/tensor_mechanics/include/kernels/WeakPlaneStress.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+class WeakPlaneStress : public Kernel
+{
+public:
+  WeakPlaneStress(const InputParameters & parameters);
+
+protected:
+  virtual Real computeQpResidual();
+  virtual Real computeQpJacobian();
+  virtual Real computeQpOffDiagJacobian(unsigned int jvar);
```
Comment (permcody) [inline modules/tensor_mechanics/include/kernels/WeakPlaneStress.h]: Let's put `override` after all of these names.

**Review by permcody — COMMENTED** [review]
This looks ok to me. I'm assign to @dschwen for approval.

File: modules/tensor_mechanics/include/kernels/WeakPlaneStress.h (hunk 1 above)
Comment (hchen139) [inline modules/tensor_mechanics/include/kernels/WeakPlaneStress.h]: Comment addressed

**Review by dschwen — APPROVED** [review]
:+1: looks good to me.

**Pull request merged**